### PR TITLE
chore: updating the swagger docs

### PR DESCRIPTION
**Pull request by gsk967**
## Description

In this pr , updating the swagger docs for uibc,ugov,intencive module queries and using statik for serving the swagger docs

added `proto-update-swagger-docs` script to update the swagger docs and generate statik.go file for swagger docs 

```bash 
# Update swagger docs and generate statik.go file
$ make proto-update-swagger-docs
```

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: swagger/statik/statik.go
[the remark this answers is not in the document]
Comment (gsk967) [reply swagger/statik/statik.go:1]: Yes, this will generated by statik

**Comment by gsk967** [comment]
> could you provide more context? Why moving to Statik?

statik allows you to embed a directory of static files into your Go binary to be later served from an http.FileSystem. 
It will fast to server the swagger docs , and cosmos and every other projects using this

**Comment by robert-zaremba** [comment]
But this pollutes this repository with binary, maybe we should do it as a part of the release? 
In other words, compile the static file when doing a release build, not as a part of regular commits.

**Comment by robert-zaremba** [comment]
let's discuss pros & cons. Maybe in the chat, to make it more efficient.

File: app/app.go [hunk 1]
```diff
@@ -145,6 +145,9 @@ import (
 	uibcoracle "github.com/umee-network/umee/v4/x/uibc/oracle"
 	uibcquota "github.com/umee-network/umee/v4/x/uibc/quota"
 	uibcquotakeeper "github.com/umee-network/umee/v4/x/uibc/quota/keeper"
+
+	// unnamed import of statik for swagger UI support
+	_ "github.com/umee-network/umee/v4/swagger/statik"
```
Comment (robert-zaremba) [inline app/app.go:150]: let's move this to tools/tools.go

File: contrib/scripts/protoc-update-swagger-docs.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# echo "SDK version ${SDK_VERSION}"
+# wget "${SDK_RAW_URL}" -O ./tmp-swagger-gen/cosmos-sdk-swagger.yaml
+
+# # download IBC swagger yaml file
+# echo "IBC version ${IBC_VERSION}"
+# wget "${IBC_RAW_URL}" -O ./tmp-swagger-gen/ibc-go-swagger.yaml
+
+SWAGGER_UI_VERSION=4.18.3
+SWAGGER_UI_DOWNLOAD_URL=https://github.com/swagger-api/swagger-ui/archive/refs/tags/v${SWAGGER_UI_VERSION}.zip
+SWAGGER_UI_PACKAGE_NAME=${SWAGGER_DIR}/swagger-ui-${SWAGGER_UI_VERSION}
```
Comment (robert-zaremba) [inline contrib/scripts/protoc-update-swagger-docs.sh:24]: what if it was already downloaded?

**Review by robert-zaremba — COMMENTED** [review]
- [ ] We need to update the documentation & readme about installing (what if a validator build himself)
- [ ] proto-update-swagger-docs must be run in release workflow (docker and goreleaser)
- [ ] need to add a note in the RELEASE NOTES

**Comment by gsk967** [comment]
@robert-zaremba  we will update release section about swagger docs when we releasing the new binary

**Review by robert-zaremba — COMMENTED** [review]
I think we also need to update the `.goreleaser.yml` or `release-umee-docker.yml`

**Comment by gsk967** [comment]
> I think we also need to update the `.goreleaser.yml` or `release-umee-docker.yml`

I added update swagger to github actions
 https://github.com/umee-network/umee/pull/2055/files#diff-0bc78b54e9cda26af1fa9b907618bfa8abca92ff587cefeeeaef62381a288fa9R50-R53 

https://github.com/umee-network/umee/pull/2055/files#diff-d067a72e796dc88a64e2cd7e755c41d17eb1812f896302fad0ad27f7858cd335R32-R36

**Pull request merged**